### PR TITLE
Header navigation, main layout FE and accounts reading lecture10

### PR DESCRIPTION
**Pull request by alialo2001**
Hello, dear mentors. I hope you are doing well! This pull request contains the implementation of the main page.

The project structure is getting messy. I wanted to ask, should I create a separate module for the  `accounts` page and store `account` and `accounts-list` components inside of it, or keep everything in the app folder? 

_Below is the current main page of a non-admin user with URL `http://localhost:4200/accounts`_

image

Thank you for checking my code and have a wonderful day!

### CODE REVIEW

**Comment by AudriusBaltusis** [comment]
> 

Please create separate folders

**Comment by AudriusBaltusis** [comment]
Please describe what was done in this PR and what HW number it is

**Comment by alialo2001** [comment]
This is the HW, where I must add the `header` and the `main page`, and I believe it is HW7. HW8 requires `static pages` and separating logic between pages, which I don't have yet.

**Comment by alialo2001** [comment]
The PR contains the implementation of the `account`, `accounts-list`, `header` and UI of the main page in general.

**Comment by alialo2001** [comment]
The HW7 is done. 

The pull request now contains:

1.    Header (with navigation)
2.  Main page UI
3. Accounts and Transactions reading from the back-end server
4. Logic separation (this is a part of the next HW, this is why the next PR will have fewer changes in the project)

File: client/src/app/transactions/transactions-list/transactions-list.component.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  selector: 'app-transactions-list',
+  templateUrl: './transactions-list.component.html',
+  styleUrls: ['./transactions-list.component.scss'],
+})
+export class TransactionsListComponent {
+  transactionSub!: Subscription;
+  transactions: ITransaction[] = [];
+
+  constructor(private userData: UserService) {
+    this.transactionSub = userData.setTransactions().subscribe(() => {
```
Comment (AudriusBaltusis) [inline client/src/app/transactions/transactions-list/transactions-list.component.ts:16]: Please unsubscribe or use `.pipe(take(1))`

**Review by AudriusBaltusis — CHANGES_REQUESTED** [review]

File: client/src/app/services/user.service.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  getAccounts(): Observable<IAccount[]> {
+    return this.http.get<IAccount[]>(
+      `http://localhost:3000/user/${localStorage.getItem('userId')}/accounts`
+    );
+  }
+
+  // to update list of transactions based on the selected account
+  private subject = new Subject<string | null>(); // a user may not have accounts
+  setAccount() {
```
[the remark this answers is not in the document]
Comment (alialo2001) [reply client/src/app/services/user.service.ts:29]: Thank you for the suggestion!

**Comment by alialo2001** [comment]
Dear Audrius, thank you again for the suggestions. 

I encountered one issue, which I couldn't fix: when I don't store `userId` in the `localStorage` and refresh the page, I lose all the data on the page (for example name in the header becomes `undefined undefined`). This is why I only implemented a helper service for the `accountId` right now. 

In addition, yesterday while I was waiting for your approval, I made some changes to the project in a different branch, but when I moved to this branch, they somehow became one branch, this is why this branch now contains HW for the next homework too, but it is not done yet. I will request your review again when I complete the following homework too. 

My apologies for all the inconvenience, and thank you for your valuable time!

**Comment by AudriusBaltusis** [comment]
Please hurry up, Friday is deadline for all HW

**Comment by alialo2001** [comment]
HW7 and HW8 are done; only the testing homework is left if approve these two HWs of course.

**Review by AudriusBaltusis — APPROVED** [review]

**Pull request merged**